gpbft/gpbft.go

```diff
@@ -243,10 +243,30 @@ func (i *instance) Receive(msg *GMessage) error {
 }
 
 func (i *instance) ReceiveAlarm() error {
+	phaseBeforeAlarm := i.phase
 	if err := i.tryCurrentPhase(); err != nil {
 		return fmt.Errorf("failed completing protocol phase: %w", err)
 	}
 
+	// Check if the alarm ended QUALITY phase, i.e. transition from QUALITY to PREPARE phase.
+	qualityPhaseEnded := phaseBeforeAlarm == QUALITY_PHASE && i.phase == PREPARE_PHASE
+	if qualityPhaseEnded {
+		// Check if there are any eligible future rounds to which to skip.
+
+		// TODO: Potential optimisations:
+		//        1) Consider preferring higher rounds to jump to first. FIP does not specify
+		//           any constraints here.
+		//        2) Build an incremental index of round candidates as messages arrive, allowing
+		//           for efficient jumping to the target round without scanning all state history.
+		for round, state := range i.rounds {
+			if round > i.round {
+				if justification, skipToRound := i.shouldSkipToRound(round, state); skipToRound {
+					i.skipToRound(round, justification)
+				}
+			}
+		}
+	}
```

Comment:
I think we can ditch this after all. See https://github.com/filecoin-project/FIPs/pull/1015

We'll do some other work elsewhere to ensure that all queued messages are processed before we try to jump. But not for this PR.

Reply:
Great; thank you for the FIP update. I think it's worth capturing the context for it in git history in form of a dedicated commit. Since commits get squashed on merge I'll open a separate PR to do this shortly after the merge of this one.